File: GXpert/GXpert.Web/Modules/ServerTypes/Playlist/ContentAssignToPlayListcontentForm.ts
import { StringEditor, EnumEditor, LookupEditor, DecimalEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EExamStatus } from "../Web/Enums.EExamStatus";
import { EPlayListContentType } from "../Web/Enums.EPlayListContentType";

export interface ContentAssignToPlayListcontentForm {
    RowIds: StringEditor;
    EContentType: EnumEditor;
    PlayListId: LookupEditor;
    ContentId: LookupEditor;
    ExamId: LookupEditor;
    LiveSessionId: LookupEditor;
    AssignmentId: LookupEditor;
    ModuleId: LookupEditor;
    EPublishStatus: EnumEditor;
    SortOrder: DecimalEditor;
}

export class ContentAssignToPlayListcontentForm extends PrefixedContext {
    static readonly formKey = 'Playlist.ContentAssignToPlayListcontentForm';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ContentAssignToPlayListcontentForm.init)  {
            ContentAssignToPlayListcontentForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = LookupEditor;
            var w3 = DecimalEditor;

            initFormType(ContentAssignToPlayListcontentForm, [
                'RowIds', w0,
                'EContentType', w1,
                'PlayListId', w2,
                'ContentId', w2,
                'ExamId', w2,
                'LiveSessionId', w2,
                'AssignmentId', w2,
                'ModuleId', w2,
                'EPublishStatus', w1,
                'SortOrder', w3
            ]);
        }
    }
}

[EPlayListContentType, EExamStatus]; // referenced types

File: GXpert/GXpert.Web/Modules/Playlist/PlayListcontent/ContentAssignToPlayListcontentForm.cs
using GXpert.Web.Enums;
using Serenity.ComponentModel;
using System;

namespace GXpert.Playlist.Forms;

[FormScript("Playlist.ContentAssignToPlayListcontentForm")]
//[BasedOnRow(typeof(PlayListContentRow), CheckNames = true)]
public class ContentAssignToPlayListcontentForm
{
    [HideOnInsert, HideOnUpdate]
    public String RowIds { get; set; }
    [HalfWidth]
    [Required]
    public EPlayListContentType? EContentType { get; set; }
   
    [LookupEditor("Playlist.PlayList"), HalfWidth]
    public int PlayListId { get; set; }
    [LookupEditor("Content.Content"), HalfWidth]

    public int ContentId { get; set; }
    [LookupEditor("Exams.Exam"), HalfWidth]
  
    public int ExamId { get; set; }
    [LookupEditor("LiveSessions.LiveSession"), HalfWidth]
    public int LiveSessionId { get; set; }
    [LookupEditor("Exams.Assignment"), HalfWidth]
    public int AssignmentId { get; set; }
    [LookupEditor("Playlist.Module"), HalfWidth]
    public int ModuleId { get; set; }
    [HalfWidth]
    [Required]
    public EExamStatus? EPublishStatus { get; set; }
    [HalfWidth]
    public float SortOrder { get; set; }
}